### PR TITLE
Several fixes for YTM and Radio

### PR DESCRIPTION
**Pull request by MarvinSchenkel**
This PR:
* Skips unavailable tracks in standard mode by verifying availability just-in-time
* Adds a warning if we suspect someone is playing music on another device
* Deduplicates radio items by `sort_name` to ensure dynamic playlists contain unique tracks

### CODE REVIEW

File: music_assistant/server/controllers/media/playlists.py [hunk 1]
```diff
@@ -319,25 +319,26 @@ async def _get_provider_dynamic_tracks(
         ]
         limit = min(limit, len(playlist_tracks))
         # use set to prevent duplicates
-        final_items = set()
```
Comment (marcelveldt) [inline music_assistant/server/controllers/media/playlists.py:322]: What's the reasoning to change from set to list ?
I have a habit of preferring sets and tuples wherever I can as its more resource friendlier

File: music_assistant/server/controllers/media/playlists.py [hunk 2]
```diff
@@ -319,25 +319,26 @@ async def _get_provider_dynamic_tracks(
...
             )
-            final_items.update(x for x in similar_tracks if x.available)
-
+            final_items.extend(x for x in similar_tracks if x.available)
+        # Remove duplicate tracks
+        radio_items = {track.sort_name: track for track in final_items}.values()
         # NOTE: In theory we can return a few more items than limit here
         # Shuffle the final items list
-        return random.sample(list(final_items), len(final_items))
+        return random.sample(list(radio_items), len(radio_items))
```
Comment (marcelveldt) [inline music_assistant/server/controllers/media/playlists.py:341]: If you're going to change from set to list, this copy to a list can be removed

File: music_assistant/server/controllers/player_queues.py [hunk 3]
```diff
@@ -631,10 +632,26 @@ async def player_ready_for_next_track(
         queue = self.get_active_queue(queue_or_player_id)
         cur_index = self.index_by_id(queue.queue_id, current_item_id)
         cur_item = self.get_item(queue.queue_id, cur_index)
-        next_index = self.get_next_index(queue.queue_id, cur_index)
-        next_item = self.get_item(queue.queue_id, next_index)
-        if not cur_item or not next_item:
-            raise QueueEmpty("No more tracks left in the queue.")
+        idx = 0
```
Comment (marcelveldt) [inline music_assistant/server/controllers/player_queues.py:635]: whats idx doing here ?

Comment (MarvinSchenkel) [reply music_assistant/server/controllers/player_queues.py:635]: > whats idx doing here ?

Good point. Forgot to add the increment, which will try to fetch the 'next next' item if the 'next' item is unplayable.

File: music_assistant/server/controllers/player_queues.py [hunk 4]
```diff
@@ -631,10 +632,25 @@ async def player_ready_for_next_track(
...
+        while True:
+            next_index = self.get_next_index(queue.queue_id, cur_index + idx)
+            next_item = self.get_item(queue.queue_id, next_index)
+            if not cur_item or not next_item:
+                raise QueueEmpty("No more tracks left in the queue.")
+            try:
+                # Check of the QueueItem is playable. For example, YT Music returns Radio Items
```
Comment (marcelveldt) [inline music_assistant/server/controllers/player_queues.py:642]:
```suggestion
                # Check if the QueueItem is playable. For example, YT Music returns Radio Items
```

**Review by marcelveldt — APPROVED** [review]

**Pull request merged**